config/samples/config.yaml

```diff
@@ -0,0 +1,26 @@
+clusterName: "example-cluster"
+timeIntervalSec: 60
+repoCredentials:
+  "https://github.com/rustyTest/testprivateRepo":
+    HTTPCredential:
+      Username: "exampleUser"
+      Password:
+        name: "http-creds"
+        key: "password"
+  "git@github.com:rustyTest/testprivateRepo.git":
+    SSHCredential:
+      SSHKey:
+        name: "ssh-creds"
+        key: "sshKey"
```

Comment:
```suggestion
  "github.com/rustyTest/testprivateRepo":
    HTTPCredential:
      Username: "exampleUser"
      Password:
        name: "http-creds"
        key: "password"
    SSHCredential:
      SSHKey:
        name: "ssh-creds"
        key: "sshKey"
```

Reply:
the prefix actually shouldn't take into account the protocol ("ssh" vs "https")

Reply:
(so you'll probably need to change your logic if it is)

Reply:
go-git throws error repo not found ,if used plane repo -`github.com/rustyTest/testprivateRepo.git` see this -https://github.com/numaproj-labs/numaplane/blob/7d398242bcdf7fb9c317c05b77fe0f4ca2b80663/internal/git/gitsync_processor_test.go#L770